packages/flutter_tools/lib/src/ios/simulators.dart

```diff
@@ -363,6 +352,57 @@ class IOSSimulator extends Device {
     return true;
   }
 
+  bool _applicationIsInstalledAndRunning(ApplicationPackage app) {
+    bool isInstalled = exitsHappy([
+      'xcrun',
+      'simctl',
+      'get_app_container',
+      'booted',
+      app.id,
+    ]);
+
+    bool isRunning = exitsHappy([
+      '/usr/bin/killall',
+      'Runner',
```

Comment:
Yeah, that part if a bit brittle. There is no way to query simctl to check if a process is running by its identifier. I am going to make the name a bit more unique to Flutter. Would need to land a change in the engine first though.